### PR TITLE
Add tests for agents deployment

### PR DESCRIPTION
**Pull request by filariow**
Add most of BDD scenarios for [agents deployment story](https://issues.redhat.com/browse/APPSVC-1288). Updates tests code.

Signed-off-by: Francesco Ilario <filario@redhat.com>

### CODE REVIEW

File: test/acceptance/features/steps/workercluster.py [hunk 1]
```diff
@@ -185,3 +371,69 @@ def ensure_worker_cluster_running(context, cluster_name: str, primaza_cluster_na
...
+    def is_not_found():
+        try:
+            worker.is_app_agent_deployed(namespace)
+        except ApiException as e:
+            return e.reason == "Not Found"
+        return False
+
+    polling2.poll(
+        target=lambda: is_not_found(),
+        step=5,
```
Comment (sadlerap) [inline test/acceptance/features/steps/workercluster.py:405]: Maybe reduce step sizes from 5 to 1?  This can _really_ slow down acceptance tests.

Comment (filariow) [reply test/acceptance/features/steps/workercluster.py:405]: that's a good idea, I'm going to apply the suggested change. We should also change this value in the other steps. I'd create a PR for that.

File: api/v1alpha1/clusterenvironment_types.go [hunk 2]
```diff
@@ -35,10 +35,10 @@ type ClusterEnvironmentSpec struct {
 	Labels []string `json:"labels,omitempty"`
 
 	// Namespaces in target cluster where applications are deployed
-	ApplicationNamespaces string `json:"applicationNamespaces,omitempty"`
+	ApplicationNamespaces []string `json:"applicationNamespaces,omitempty"`
 
 	// Namespaces in target cluster where services are discovered
-	ServiceNamespaces string `json:"serviceNamespaces,omitempty"`
+	ServiceNamespaces []string `json:"serviceNamespaces,omitempty"`
```
Comment (baijum) [inline api/v1alpha1/clusterenvironment_types.go:41]: Can we make these API changes in a separate PR and merge it fast?

Comment (filariow) [reply api/v1alpha1/clusterenvironment_types.go:41]: that's a good idea, I opened the following PRs:
- #38 
- #39

**Comment by filariow** [comment]
includes changes from #38 and #39

**Review by dperaza4dustbit — APPROVED** [review]
/lgtm

**Review by dperaza4dustbit — APPROVED** [review]
/lgtm
